src/js/bin/index.js

```diff
@@ -0,0 +1,266 @@
+            confirmProjectCreation(directoryName, packageConfig);
+        });
+};
+
+const confirmProjectCreation = (directoryName, packageConfig) => {
+    const question = `${JSON.stringify(packageConfig, null, 2)}\n\nIs that okay? ${chalk.grey('yes')}`;
+
+    rl.question(question, answer => {
+        if (answer) {
+            promptPackageJson();
+        } else {
+            createProject(directoryName, packageConfig);
+            rl.close();
+        }
+    });
+};
+
+const printValidationResults = results => {
+    if (typeof results !== 'undefined') {
+        results.forEach(error => {
+            console.error(chalk.red(`  *  ${error}`));
+        });
+    }
+};
+
+const shouldBeIgnored = src => {
+    for (var i = 0; i < ignoreFiles.length; i++) {
+        if (src.indexOf(ignoreFiles[i]) === 0) {
+            return true;
+        }
+    }
+    return false;
+};
+
+const createProject = (name, config) => {
+    const targetDir = path.resolve(name);
+    const sourceDir = path.resolve(__dirname, '../../../');
+    const appName = path.basename(targetDir);
+
+    console.log();
+    console.log('👍🏻  Roger that.');
+    console.log();
+
+    const spinner = new Spinner('%s Please wait while we`re bootstrapping your project');
+    spinner.setSpinnerString('⠋⠙⠹⠸⠼⠴⠦⠧⠇⠏');
+    spinner.start();
+
+    const newPackageJson = Object.assign({}, packageJson, packageConfig);
+
+    ignorePackageConfig.forEach(ignore => {
+        delete newPackageJson[ignore];
+    });
+    fs
+        .emptyDir('./tmp')
+        .then(() => {
+            console.log();
+            console.log('🚚  Fetching project files from repository');
+            return git.Clone('https://github.com/researchgate/node-package-blueprint.git', './tmp');
+        })
+        .then(() => {
+            console.log();
+            console.log('📦  Prepare project');
+            return fs.copy(`./tmp`, targetDir, {
+                filter: (src, dest) => {
+                    const relativePath = path.relative(`./tmp`, src);
+                    return !shouldBeIgnored(relativePath);
+                },
+            });
+        })
+        .then(() => {
+            return fs.outputFile(path.join(targetDir, 'package.json'), JSON.stringify(newPackageJson, null, 2));
+        })
+        .then(() => timeout())
+        .then(() => {
+            return fs.copy(path.resolve(__dirname, 'template'), targetDir);
+        })
+        .then(() => {
+            console.log();
+            console.log();
+            console.log(`🙌🏻  Your project is ready and is waiting for you over here: ${targetDir}`);
+            console.log();
+            spinner.stop();
+        });
+};
+
+const timeout = () =>
+    new Promise(resolve => {
+        setTimeout(resolve, 1000);
+    });
+
+const isNameValid = name => {
+    const validationResult = validateProjectName(name);
+    if (!validationResult.validForNewPackages) {
+        console.error(
```

Comment:
`throw new Error()` works along with the `.catch()` to create a single error entry point.